apps/antalmanac/src/components/dialogs/Import.tsx

```diff
@@ -0,0 +1,264 @@
+import { useCallback, useState } from 'react';
+import { ProviderContext, useSnackbar } from 'notistack';
+import {
+    Box,
+    Button,
+    Dialog,
+    DialogContent,
+    type DialogProps,
+    DialogTitle,
+    InputLabel,
+    Link,
+    Stack,
+    Tab,
+    Tabs,
+    TextField,
+    Typography,
+} from '@mui/material';
+import RightPaneStore from '$components/RightPane/RightPaneStore';
+import TermSelector from '$components/RightPane/CoursePane/SearchForm/TermSelector';
+import AppStore from '$stores/AppStore';
+import { addCoursesMultiple, getCourseInfo, queryWebsoc, queryZotCourse } from '$lib/helpers';
+import { addCustomEvent } from '$actions/AppStoreActions';
+import analyticsEnum, { logAnalytics } from '$lib/analytics';
+
+interface NestedFormProps {
+    onClose?: () => unknown;
+}
+
+interface ImportCoursesProps {
+    enqueueSnackbar: ProviderContext['enqueueSnackbar'];
+    term: string;
+    sectionCodes: string[];
+}
+
+async function importCourses(props: ImportCoursesProps) {
+    const { enqueueSnackbar, term, sectionCodes } = props;
+
+    const currentScheduleIndex = AppStore.getCurrentScheduleIndex();
+
+    try {
+        const websocData = await queryWebsoc({ term, sectionCodes: sectionCodes.join(',') });
+        const courseInfo = getCourseInfo(websocData);
+        const sectionsAdded = addCoursesMultiple(courseInfo, term, currentScheduleIndex);
+
+        logAnalytics({
+            category: analyticsEnum.nav.title,
+            action: analyticsEnum.nav.actions.IMPORT_STUDY_LIST,
+            value: sectionsAdded / (sectionCodes.length || 1),
+        });
+
+        if (sectionsAdded === sectionCodes.length) {
+            enqueueSnackbar(`Successfully imported ${sectionsAdded} of ${sectionsAdded} classes!`, {
```

Comment:
We should try to migrate away from the `AppStore`'s notification handler, and just use `enqueueSnackbar` directly for simplicity.